crates/pop-cli/src/common/contracts.rs

```diff
@@ -186,6 +186,39 @@ pub(crate) async fn map_account(
 	Ok(())
 }
 
+/// Resolves the signer for contract operations (deployment or calls).
```

Comment:
I don't like this function as it is right now. Reasons:

- Takes assumptions on parameters passed in a given workflow, yet the function itself is supposed to be agnostic.
- Mutates received parameters. That might have been an option in C's old days when you could not return tuples or enums, but I definitely wouldn't do that with Rust's capabilities.

My proposal is to make this function internal inside `UpContractCommand` and take a mutable reference to itself, and only them mutate its state, instead of delegating that functionality to a foreign function.